File: Backend/Microservices/Commodity/Eefa.Commodity.Application/Commands/CommodityCategory/Update/UpdateCommodityCategoryCommand.cs
using AutoMapper;
using Eefa.Common.CommandQuery;
using Eefa.Common;
using Eefa.Common.Data;
using MediatR;
using System.Threading;
using System.Threading.Tasks;
using Eefa.Commodity.Application.Queries.Commodity;

namespace Eefa.Commodity.Application.Commands.CommodityCategory.Update
{
    public class UpdateCommodityCategoryCommand : CommandBase, IRequest<ServiceResult<CommodityCategoryModel>>, IMapFrom<Data.Entities.CommodityCategory>, ICommand
    {
        public int Id { get; set; }

        /// <summary>
        /// کد والد
        /// </summary>
        public int? ParentId { get; set; }

        /// <summary>
        /// کد سطح
        /// </summary>
        public string LevelCode { get; set; } = default!;

        /// <summary>
        /// کد
        /// </summary>
        public string Code { get; set; } = default!;

        public int CodingMode { get; set; } = default!;

        /// <summary>
        /// نام اختصاصی 
        /// </summary>
        public string? UniqueName { get; set; }

        /// <summary>
        /// عنوان
        /// </summary>
        public string? Title { get; set; }

        /// <summary>
        /// کدواحد اندازه گیری
        /// </summary>
        public int? MeasureId { get; set; } = default!;

        /// <summary>
        /// ترتیب نمایش
        /// </summary>
        public int OrderIndex { get; set; } = default!;

        /// <summary>
        /// آیا فقط قابل خواندن است؟
        /// </summary>
        public bool IsReadOnly { get; set; } = default!;

        public bool? RequireParentProduct { get; set; }

        public void Mapping(Profile profile)
        {
            profile.CreateMap<UpdateCommodityCategoryCommand, Eefa.Commodity.Data.Entities.CommodityCategory>()
                .IgnoreAllNonExisting();
        }
    }


    public class UpdateCommodityCategoryCommandHandler : IRequestHandler<UpdateCommodityCategoryCommand, ServiceResult<CommodityCategoryModel>>
    {
        private readonly IRepository<Data.Entities.CommodityCategory> _repository;
        private readonly IMapper _mapper;

        public UpdateCommodityCategoryCommandHandler(IRepository<Data.Entities.CommodityCategory> repository, IMapper mapper)
        {
            _mapper = mapper;
            _repository = repository;
        }

        public async Task<ServiceResult<CommodityCategoryModel>> Handle(UpdateCommodityCategoryCommand request, CancellationToken cancellationToken)
        {
            var entity = await _repository.Find(request.Id);
            request.MeasureId = request.MeasureId == 0 ? null : request.MeasureId;

            _mapper.Map<UpdateCommodityCategoryCommand, Data.Entities.CommodityCategory>(request, entity);
            _repository.Update(entity);
            await _repository.SaveChangesAsync(cancellationToken);

            return ServiceResult<CommodityCategoryModel>.Success(
               _mapper.Map<CommodityCategoryModel>(entity)
           );
        }
    }
}

File: Frontend/angular-app/src/app/modules/commodity/repositories/commodity-category/commands/update-commodity-category-command.ts
import {Inject} from "@angular/core";
import {ServiceResult} from "../../../../../core/models/service-result";
import {IRequest, IRequestHandler} from "../../../../../core/services/mediator/interfaces";
import {HttpService} from "../../../../../core/services/http/http.service";
import {ApplicationError} from "../../../../../core/exceptions/application-error";
import {MediatorHandler} from "../../../../../core/services/mediator/decorator";
import {NotificationService} from "../../../../../shared/services/notification/notification.service";
import {ValidationRule} from "../../../../../core/validation/validation-rule";
import {HttpRequest} from "../../../../../core/services/http/http-request";
import {CommodityCategory} from "../../../entities/commodity-category";

export class UpdateCommodityCategoryCommand extends IRequest<UpdateCommodityCategoryCommand, CommodityCategory> {
  public id: number | undefined = undefined
  public parentId: number | undefined = undefined;
  public levelCode: string | undefined = undefined;
  public code: string | undefined = undefined;
  public codingMode: number | undefined = undefined;
  public title: string | undefined = undefined;
  public measureId: number | undefined = undefined;
  public orderIndex: number | undefined = undefined;
  public isReadOnly: boolean | undefined = undefined;
  public requireParentProduct: boolean | undefined = undefined;

  constructor() {
    super();
  }

  mapFrom(entity: CommodityCategory): UpdateCommodityCategoryCommand {
    this.mapBasics(entity,this)
    return this;
  }

  mapTo(): CommodityCategory {
    throw new ApplicationError(UpdateCommodityCategoryCommand.name, this.mapTo.name, 'Not Implemented Yet')
  }

  get url(): string {
    return "/commodity/commodityCategory/update";
  }

  get validationRules(): ValidationRule[] {
    return [];
  }
}

@MediatorHandler(UpdateCommodityCategoryCommandHandler.name)
export class UpdateCommodityCategoryCommandHandler implements IRequestHandler<UpdateCommodityCategoryCommand, CommodityCategory> {
  constructor(
    @Inject(HttpService) private _httpService: HttpService,
    @Inject(NotificationService) private _notificationService: NotificationService,
  ) {
  }

  async Handle(request: UpdateCommodityCategoryCommand): Promise<CommodityCategory> {
    let httpRequest: HttpRequest<UpdateCommodityCategoryCommand> = new HttpRequest<UpdateCommodityCategoryCommand>(request.url, request);
    // httpRequest.Query  += ;
    // httpRequest.Headers = httpRequest.Headers.append()

    return await this._httpService.Put<UpdateCommodityCategoryCommand, ServiceResult<CommodityCategory>>(httpRequest).toPromise().then(response => {
      this._notificationService.showSuccessMessage()
      return response.objResult
    })
  }
}
